src/main/java/bio/terra/cda/app/model/NestedColumn.java

```diff
@@ -0,0 +1,58 @@
+package bio.terra.cda.app.model;
+
+/**
+ * Table cells that represent as nested structures must be unnested from it decendant heiarchy. the
+ * UNNEST clause is generated by unwinding the "dot" hierarchy such that:
+ * gdc-bg-sample.cda_mvp.v3.ResearchSubject.identifier.value would unwind to the following. "SELECT
+ * DISTINCT (_D.value) FROM gdc-bq-sample.cda_mvp.v3, UNNEST(ResearchSubject) as _C,
+ * UNNEST(_C.identifier) AS _D";
+ */
+public class NestedColumn {
+  private String column;
+  private String unnestClause;
+
+  public NestedColumn(String columnName) {
+    String[] c = columnName.split("\\.");
+    generateUnnestClause(columnName);
+  }
+
+  public String getColumn() {
+    return column;
+  }
+
+  public void setColumn(String column) {
+    this.column = column;
+  }
+
+  public String getUnnestClause() {
+    return unnestClause;
+  }
+
+  public void setUnnestClause(String unnestClause) {
+    this.unnestClause = unnestClause;
+  }
+
+  // column ->
+  // SELECT DISTINCT(column)
+
+  // D.column ->
+  // SELECT DISTINCT(_D.column) FROM TABLE, UNNEST(D) AS _D
+
+  // A.B.C.D.column ->
+  // # SELECT DISTINCT(_D.column) FROM TABLE, UNNEST(A) AS _A, UNNEST(_A.B) AS _B, UNNEST(_B.C) AS
+  // _C, UNNEST(_C.D) AS _D
+  private void generateUnnestClause(String columnName) {
```

Comment:
This method is complex enough that it needs a unit test. You can unit test at the `NestedColumn()` level, e.g. create a `NestedColumn()` and assert that the `column` and `unnestClause` fields have the expected values.

Reply:
Wrote a static Builder class that returns a NestedColumn object
That class has Unit testing implemented